pkg/daemon/ceph/osd/volume.go

```diff
@@ -719,73 +728,111 @@ func (a *OsdAgent) initializeDevicesLVMMode(context *clusterd.Context, devices *
 		}
 	}
 
+	var prepareArgs []string
+	if hasMetadataPartition {
+		// ceph-volume lvm prepare --data {vg/lv} --block.wal {partition} --block.db {/path/to/device}
+		baseArgs := []string{"-oL", cephVolumeCmd, "--log-path", logPath, "lvm", "prepare", storeFlag}
+		if a.storeConfig.EncryptedDevice {
+			baseArgs = append(baseArgs, encryptedFlag)
+		}
+		prepareArgs = baseArgs
+	}
+
 	for md, conf := range metadataDevices {
 
 		mdArgs := batchArgs
-		if _, ok := conf["osdsperdevice"]; ok {
-			mdArgs = append(mdArgs, []string{
-				osdsPerDeviceFlag,
-				conf["osdsperdevice"],
-			}...)
+		if hasMetadataPartition {
+			mdArgs = prepareArgs
+		} else {
+			if _, ok := conf["osdsperdevice"]; ok {
+				mdArgs = append(mdArgs, []string{
+					osdsPerDeviceFlag,
+					conf["osdsperdevice"],
+				}...)
+			}
 		}
 		if _, ok := conf["deviceclass"]; ok {
 			mdArgs = append(mdArgs, []string{
 				crushDeviceClassFlag,
 				conf["deviceclass"],
 			}...)
 		}
-		if _, ok := conf["databasesizemb"]; ok {
+		if hasMetadataPartition {
```

Comment:
When using part and the original non-part as metadata at the same time, the non-part may be executed in the part method. It is better to put `hasMetadataPartition` into `metadataDevices`, or add comment for this situation.

Reply:
done